libraries/core_libs/network/src/tarcap/packets_handlers/get_pbft_sync_packet_handler.cpp

```diff
@@ -79,10 +79,22 @@ void GetPbftSyncPacketHandler::sendPbftBlocks(dev::p2p::NodeID const &peer_id, s
     }
 
     dev::RLPStream s;
-    s.appendList(3);
-    s << (pbft_chain_synced && last_block);
-    s << last_block;
-    s.appendRaw(data);
+    if (pbft_chain_synced && last_block) {
+      s.appendList(4);
+      s << (pbft_chain_synced && last_block);
+      s << last_block;
+      s.appendRaw(data);
+      const auto votes = db_->getLastBlockCertVotes();
```

Comment:
again, this could be part of pbft managers memory

Reply:
I guess other guys can weigh in but I am pretty sure that rocksdb memory caching does exactly the same thing so if db provides memory caching of data I do not see why we need to duplicate it and implement one in our code.

Reply:
One thing that we do know that in our process it is just rocksdb caching that is taking most of the RAM. Caching can be configured in rocksdb globally and per column. By introducing our own caches, we are basically doubling the memory usage and we gain no performance. In my opinion we should either:
1. Use rocksdb memory caching especially for small columns unless we can find a specific reason why we would cache data better than rocksdb
2. This is something I would not recommend but turn off rocksdb caching and have our own caching mechanism. Having both on does not make sense to me

Reply:
@MatusKysel I leave it to you as it is your orig. comment.

Reply:
If this won't be fixed in this PR let's create task for it, otherwise we will forget  and we could gain some performance here

Reply:
Db usually also have write-through caching. When you write data to rocksdb there is actually a setting where you can choose if you wait for the call to return even if the data is not written to the disk yet. It is written to memory cache on write as well.

Reply:
I know, I in past I have made some test for trxs/dags and this cache is still like 10x slower then access to memory

Reply:
So I did a quick test. I save 1000 votes in the column. And then I test 3 scenarios:
1. Reading all 1000 votes from database which includes rlp deserialization and this took 14139µs
2. Reading 1000 votes from local memory but also from vector or serialized rlp data and this took 13684µs
3. Simple copy of 1000 votes from memory took 250µs

Based on this I would say that reading from rocksdb memory cache is as fast as reading from local memory but the main bottleneck is rlp serialization. 
Based on this it would be much faster to keep local copy of votes so I can implement that but if we are going for some great performance maybe we should consider if storing data as rlp is efficient and consider some other faster serialization format

Reply:
I created issue to check other serialization formats https://github.com/Taraxa-project/taraxa-node/issues/1799. If you feel like you resolved the other part of the comment(having these votes in memory), I think you can resolve this and merge without as he is not working today and qi is waiting for this PR

Reply:
@MatusKysel I will merge and if you feel this is still unresolved create an issue with the details on how it should be improved